pop/app.go

```diff
@@ -0,0 +1,377 @@
+package main
+
+import (
+	"encoding/base64"
+	"errors"
+	"os"
+	"path"
+
+	"github.com/dedis/cothority/cosi/check"
+	_ "github.com/dedis/cothority/cosi/protocol"
+	_ "github.com/dedis/cothority/cosi/service"
+
+	"fmt"
+	"io/ioutil"
+
+	"net"
+
+	"strings"
+
+	"bytes"
+
+	"github.com/BurntSushi/toml"
+	"github.com/dedis/cothority/pop/service"
+	"gopkg.in/dedis/crypto.v0/abstract"
+	"gopkg.in/dedis/crypto.v0/anon"
+	"gopkg.in/dedis/crypto.v0/config"
+	"gopkg.in/dedis/crypto.v0/random"
+	"gopkg.in/dedis/onet.v1/app"
+	"gopkg.in/dedis/onet.v1/crypto"
+	"gopkg.in/dedis/onet.v1/log"
+	"gopkg.in/dedis/onet.v1/network"
+	"gopkg.in/urfave/cli.v1"
+)
+
+func init() {
+	network.RegisterMessage(Config{})
+}
+
+// Config represents either a manager or an attendee configuration.
+type Config struct {
+	// Index of the attendee in the final statement. If the index
+	// is -1, then this pop holds an organizer.
+	Index int
+	// Private key of attendee or organizer, depending on value
+	// of Index.
+	Private abstract.Scalar
+	// Public key of attendee or organizer, depending on value of
+	// index.
+	Public abstract.Point
+	// Address of the linked conode.
+	Address network.Address
+	// Final statement of the party.
+	Final *service.FinalStatement
+	// config-file name
+	name string
+}
+
+func main() {
+	appCli := cli.NewApp()
+	appCli.Name = "Proof-of-personhood party"
+	appCli.Usage = "Handles party-creation, finalizing, pop-token creation, and verification"
+	appCli.Version = "0.1"
+	appCli.Commands = []cli.Command{}
+	appCli.Commands = []cli.Command{
+		commandOrg,
+		commandClient,
+		{
+			Name:      "check",
+			Aliases:   []string{"c"},
+			Usage:     "Check if the servers in the group definition are up and running",
+			ArgsUsage: "group.toml",
+			Action: func(c *cli.Context) error {
+				return check.Config(c.Args().First(), false)
+			},
+		},
+	}
+	appCli.Flags = []cli.Flag{
+		cli.IntFlag{
+			Name:  "debug,d",
+			Value: 0,
+			Usage: "debug-level: 1 for terse, 5 for maximal",
+		},
+		cli.StringFlag{
+			Name:  "config,c",
+			Value: "~/.config/cothority/pop",
```

Comment:
Not for now, but what do you think about having some kind of utility in onet to return the "good" configuration folder ? `~/.config` is very linux, but I remember you saying that Mac is more USers/Library or stg ?